terratorch/models/backbones/prithvi_vit.py

```diff
@@ -81,7 +83,7 @@ def _create_prithvi(
     if "features_only" in kwargs:
         kwargs = {k: v for k, v in kwargs.items() if k != "features_only"}
 
-    kwargs["in_chans"] = len(model_bands)
+    kwargs["in_chans"] = _estimate_in_chans(model_bands=model_bands)
```

Comment:
If the user passes `model_bands` with intervals, don't we need to "expand" this into the full form in order to use it in the `checkpoint_filter_fn`? I think this part is missing.

Then, once we have expanded, we should just be able to call `len` on that. I think we shouldn't need this `_estimate_in_chans`, the above approach should be simpler.

Reply:
Maybe the modification from the last commits solves this issue.